config/config.go

```diff
@@ -1,156 +1,93 @@
 	DefaultCorpusDir = "out/corpus"
 
 	// DefaultProjectDir is the directory where the project is located.
 	DefaultProjectDir = "out/project"
-
-	// DefaultReportName is the directory name where fuzzing results are
-	// stored.
-	DefaultReportName = "fuzz_results"
 )
 
-// Config holds the configuration parameters for the fuzzing setup.
+// Config encapsulates all configuration parameters required for the fuzzing
+// system. It is populated from environment variables and command-line flags.
+//
+//nolint:lll
 type Config struct {
-	// ProjectSrcPath is the Git repository URL of the project to be fuzzed.
-	ProjectSrcPath string
+	ProjectSrcPath string `long:"project_src_path" description:"Git repo URL of the project to fuzz" required:"true" env:"PROJECT_SRC_PATH"`
 
-	// GitStorageRepo is the Git repository where the input corpus is
-	// stored.
-	GitStorageRepo string
+	GitStorageRepo string `long:"git_storage_repo" description:"Git repository where the input corpus is stored" required:"true" env:"GIT_STORAGE_REPO"`
 
-	// Path to store fuzzing results, relative to the current working
-	// directory
-	FuzzResultsPath string
+	FuzzResultsPath string `long:"fuzz_results_path" description:"Path to store fuzzing results" env:"FUZZ_RESULTS_PATH" required:"true"`
 
-	// FuzzPkgs are the specific Go packages within the repository that will
-	// be fuzzed.
-	FuzzPkgs []string
+	FuzzPkgs []string `long:"fuzz_pkg" description:"Comma-separated list of package names to fuzz" required:"true" env:"FUZZ_PKG" env-delim:","`
 
-	// FuzzTime is the duration (in seconds) for which the fuzzing engine
-	// should run.
-	FuzzTime string
+	FuzzTime time.Duration `long:"fuzz_time" description:"Duration in seconds for fuzzing run" env:"FUZZ_TIME" default:"120s"`
 
-	// NumProcesses specifies the number of fuzzing processes to run
-	// concurrently.
-	NumProcesses int
+	NumProcesses int `long:"num_processes" description:"Number of concurrent fuzzing processes" env:"FUZZ_NUM_PROCESSES" default:"1"`
 }
 
-// LoadEnv loads environment variables from a .env file in the current
-// directory. It should be called before accessing any environment variables in
-// the application. If the file is not found or fails to load, the application
-// will return the error.
-func LoadEnv() error {
-	// If '.env' does not exist, load environment variables from the
-	// process's environment
-	if _, err := os.Stat(".env"); os.IsNotExist(err) {
-		return nil
-	}
+// LoadConfig parses configuration from environment variables and command-line
+// flags. It validates required fields and applies sensible defaults.
+// Returns a pointer to a Config struct or an error if validation fails.
+func LoadConfig() (*Config, error) {
+	var cfg Config
 
-	// load environment variables from a .env file
-	if err := godotenv.Load(); err != nil {
-		return fmt.Errorf("failed to load .env file: %w", err)
+	// Parse configuration, populating the cfg struct.
+	if _, err := flags.Parse(&cfg); err != nil {
```

Comment:
I think we should also support loading the flags from a config file.  I imagine that will be the preferred way to run.